tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,253 @@
+        if not coprs:
+            return
+        # Try to install copr plugin
+        self.debug('Make sure the copr plugin is available.')
+        try:
+            guest.execute(f'rpm -q {plugin} || {command} install -y {plugin}')
+        # Enable repositories manually for epel6
+        except tmt.utils.RunError:
+            for copr in coprs:
+                self.info('copr', copr, 'green')
+                self.enable_copr_epel6(copr, guest)
+        # Enable repositories using copr plugin
+        else:
+            for copr in coprs:
+                self.info('copr', copr, 'green')
+                guest.execute(f'{command} copr enable -y {copr}')
+
+    def prepare_install_local(self, step):
+        """ Copy packages to the test system """
+        self.rpms_directory = os.path.join(step.workdir, 'rpms')
+        os.makedirs(self.rpms_directory)
+
+        # Copy local packages into workdir, push to guests
+        for package in self.local_packages:
+            self.verbose(os.path.basename(package), shift=1)
+            self.debug(f"Copy '{package}' to '{self.rpms_directory}'.", level=3)
+            shutil.copy(package, self.rpms_directory)
+        for guest in step.plan.provision.guests():
+            guest.push()
+
+    def install_debuginfo(self):
+        """ Install debuginfo packages """
+        packages = self.prepare_packages(
+            self.debuginfo_packages, title="debuginfo")
+        # Make sure debuginfo-install is present on the target system
+        self.guest.execute(f"{self.command} install -y /usr/bin/debuginfo-install")
+        self.guest.execute(f"debuginfo-install -y {packages}")
+
+    def install(self, step):
+        """ Perform the actual installation """
+        if len(self.local_packages) > 0:
+            self.prepare_install_local(step)
+            self.install_local()
+        if len(self.repo_packages) > 0:
+            self.install_from_repository()
+        if len(self.debuginfo_packages) > 0:
+            self.install_debuginfo()
+
+
+class InstallDnf(InstallBase):
+    """ Install packages using dnf or yum """
+
+    def install_local(self):
+        """Install copied local packages"""
+        # Use both dnf install/reinstall to get all packages refreshed
+        # FIXME Simplify this once BZ#1831022 is fixed/implemeted.
+        self.guest.execute(f"{self.command} install {self.options} {self.rpms_directory}/*")
+        self.guest.execute(f"{self.command} reinstall {self.options} {self.rpms_directory}/*")
+        summary = fmf.utils.listed(self.local_packages, 'local package')
+        self.info('total', f"{summary} installed", 'green')
+
+    def install_from_repository(self):
+        """ Install packages from the repository """
+        packages = self.prepare_packages(self.repo_packages, title="package")
+        # Extra ignore/check for yum to workaround BZ#1920176
+        check = f'rpm -q --whatprovides {packages}'
+        if 'yum' in self.command:
```

Comment:
I'd rather check `self.pkg_mgr`, that should be either `yum` or `dnf`, we'd be checking the source of truth from which `self.command` was derived.